### PR TITLE
Fix simple implicit type conversions

### PR DESCRIPTION
**Pull request by TuningYourCode**
My first contribute to a c++ project - hopefully quiet some "useful".

Only changed places which seem to be local variables and were using wrong type. 

My IDE (Clion - JetBrains) does no longer complain about incorrect type of these variables, building works and application was tested for some minutes with different tasks including wacom tablet.

If my contribution is ok for you i could also try to do more major stuff like easy issues or fixing some of the other implicit type conversions. 

For me as beginner to this project it is quiet strange why page-related variables (count, current page id etc.) are (signed) integer instead of size_t which should be on all common plattforms at least that big like int and will show any errors/tips in IDEs.

### CODE REVIEW

**Comment by mjg** [comment]
Fixing these warnings is certainly nice because it helps catch real "errors". I'm wondering, though, whether `auto` is the more readable solution for most of these. The codebase is "organically grown", so you will see different styles. But, e.g. the local variable in a `for` loop is a good candidate for `auto` (unless you go full throttle and convert to range loops).

**Comment by Febbe** [comment]
> I'm wondering, though, whether auto is the more readable solution for most of these.

This is indeed an opinionated question and there is no correct answer to it, I can only throw in my opinion: Using auto everywhere is better, since it prevents many casting errors. Also as maintainer, I want to prove the logic of the implementation and not the semantic. Using auto means for me, that I don't need to bother, if the type is correct. There is also a blog post of Herb Sutter to this: https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/

Other maintainers here have of course another opinion. Every one of us can handle the AAA-Style and the opposite, only if it's too unreadable for the reviewer, you must change this. Often it is marked as **nitpick**.

**Comment by Febbe** [comment]
Very good, we will merge this after we released version 1.1.0. This was also on my to-do list.

**Comment by rolandlo** [comment]
@TuningYourCode Can you please rebase on current master and solve the merge conflicts?

File: src/model/Document.cpp
Comment (rolandlo) [inline src/model/Document.cpp:261]:
```suggestion
    if (page != npos) {
```

File: src/undo/InsertDeletePageUndoAction.cpp
Comment (rolandlo) [inline src/undo/InsertDeletePageUndoAction.cpp:73]:
```suggestion
    if (pNr == npos) {
```

File: src/undo/PageBackgroundChangedUndoAction.cpp
Comment (rolandlo) [inline src/undo/PageBackgroundChangedUndoAction.cpp:34]:
```suggestion
    if (pageNr == npos) {
```

File: src/undo/PageBackgroundChangedUndoAction.cpp
Comment (rolandlo) [inline src/undo/PageBackgroundChangedUndoAction.cpp:60]:
```suggestion
    if (pageNr == npos) {
```

**Review by rolandlo — CHANGES_REQUESTED** [review]
I have some nitpicks, see in the code. The rest looks good to me.

**Comment by rolandlo** [comment]
Anything else you'd like to address in this PR? Or is it ready to be merged from your side?

**Comment by TuningYourCode** [comment]
Feel free to merge it :) If i find time again i will just open another PR.

**Comment by rolandlo** [comment]
Merging in 24 hours if no objections are raised.

**Pull request merged**
